src/SFA.DAS.LoginService.Web/Infrastructure/IdentityServerStartupExtensions.cs

```diff
@@ -63,4 +75,19 @@ public static void AddIdentityServer(this IServiceCollection services, ILoginCon
             }
         }
     }
+
+    public interface IManagedIdentityTokenProvider
+    {
+        string GetAccessToken();
+    }
+
+    public class ManagedIdentityAzureTokenProvider : IManagedIdentityTokenProvider
+    {
+        public string GetAccessToken()
+        {
+            return new AzureServiceTokenProvider()
+                .GetAccessTokenAsync("https://database.windows.net/")
+                .GetAwaiter().GetResult();
+        }
```

Comment:
why can't we write this function in the usual async pattern?  Please ignore, I can see why